usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/datacontainers/DateTime.java

```diff
@@ -121,98 +162,278 @@ private boolean isLeapYear(int year) {
+        if (hour < FIRST_TWO_DIGITS) {
+            dateTime.append(ZERO_DIGIT_CHAR);
+        }
+        dateTime.append(hour).append(TIME_SEPARATOR_CHAR);
+        if (minutes < FIRST_TWO_DIGITS) {
+            dateTime.append(ZERO_DIGIT_CHAR);
+        }
+        dateTime.append(minutes).append(TIME_SEPARATOR_CHAR);
+        if (seconds < FIRST_TWO_DIGITS) {
+            dateTime.append(ZERO_DIGIT_CHAR);
+        }
+        dateTime.append(seconds);
+        return dateTime.toString();
+    }
+
+    @Override
+    public int compareTo(DateTime dateTime) {
+        if (year != dateTime.year) {
+            return year - dateTime.year;
+        }
+        if (month != dateTime.month) {
+            return month - dateTime.month;
+        }
+        if (day != dateTime.day) {
+            return day - dateTime.day;
+        }
+        if (hour != dateTime.hour) {
+            return hour - dateTime.hour;
+        }
+        if (minutes != dateTime.minutes) {
+            return minutes - dateTime.minutes;
+        }
+        return seconds - dateTime.seconds;
+    }
+
+    public static class Builder {
+
+        /**
+         * Build a DateTime with all parameters.
+         * @param year The year
+         * @param month The month
+         * @param day The day
+         * @param hour The hour
+         * @param minutes The minutes
+         * @param seconds The seconds
+         * @return The built DateTime
+         * @throws InvalidFormatException The format is not valid
+         */
+        public static DateTime build(int year, int month, int day, int hour, int minutes, int seconds)
+            throws InvalidFormatException {
+            return new DateTime(year, month, day, hour, minutes, seconds);
+        }
+
+        /**
+         * Build a DateTime with just the date.
+         * @param year The year
+         * @param month The month
+         * @param day The day
+         * @return The built DateTime with just the date
+         * @throws InvalidFormatException The format is not valid
+         */
+        public static DateTime build(int year, int month, int day)
+            throws InvalidFormatException {
+            return new DateTime(year, month, day, 0, 0, 0);
+        }
+
+        /**
+         * Build a DateTime with the format yyyy-MM-d'T'hh:mm:ss (apostrophes not included).
+         * @param date The date with full string format
+         * @return The built DateTime with full string format
+         */
+        public static DateTime buildFullString(String date) {
+            return new DateTime(date, true);
+        }
+
+        /**
+         * Build a DateTime with the format yyyy-MM-d.
+         * @param date The date with date string format
+         * @return The built DateTime with date string format
+         */
+        public static DateTime buildDateString(String date) {
+            return new DateTime(date, false);
+        }
+
+        /**
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [First sentence should end with a period.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5351293)